resolwe/flow/tests/test_access_api.py

```diff
@@ -0,0 +1,151 @@
+        self.contributor = user_model.objects.create_user('test_user', 'test_pwd')
+
+    def tearDown(self):
+        for data in Data.objects.all():
+            data_dir = os.path.join(settings.FLOW_EXECUTOR['DATA_DIR'], str(data.id))
+            shutil.rmtree(data_dir, ignore_errors=True)
+
+    def test_access_api_from_local_executor_process(self):  # pylint: disable=invalid-name
+        """Test if a process running via local executor can access API."""
+        # create a test collection that we'll access via a test process
+        collection_name = 'collection-foo'
+        collection = Collection.objects.create(
+            name=collection_name,
+            contributor=self.contributor,
+        )
+        assign_perm("view_collection", AnonymousUser(), collection)
+
+        # create a test process that will obtain information about the test
+        # collection via API using curl
+        process = Process.objects.create(
+            name='Test accessing API from process',
+            requirements={'expression-engine': 'jinja'},
+            contributor=self.contributor,
+            type='data:test:api-access',
+            input_schema=[],
+            output_schema=[
+                {'name': 'collection-list', 'type': 'basic:json:'},
+            ],
+            run={
+                'language': 'bash',
+                'program': """
+re-save collection-list "$(curl $RESOLWE_API_HOST/collection)"
+"""
+            }
+        )
+
+        data = Data.objects.create(
+            name='data-bar',
+            contributor=self.contributor,
+            process=process,
+            input={},
+        )
+
+        # run executor
+        manager.communicate(verbosity=0)
+
+        # update data object
+        data.refresh_from_db()
+
+        # check if information about the test collection obtained via API is
+        # correct
+        collection_list = Storage.objects.get(data=data).json
+        self.assertEqual(len(collection_list), 1)
+        self.assertEqual(collection_list[0]['slug'], collection_name)
+
+
+flow_executor_settings = copy.copy(  # pylint: disable=invalid-name
+    getattr(settings, 'FLOW_EXECUTOR', {}))
+flow_executor_settings['NAME'] = 'resolwe.flow.executors.docker'
+flow_executor_settings['CONTAINER_IMAGE'] = 'centos'
+
+
+@unittest.skipUnless(*check_docker())
+@override_settings(FLOW_EXECUTOR=flow_executor_settings)
+class AccessAPIDockerExecutorTestCase(LiveServerTestCase):
+
+    def setUp(self):
+        user_model = get_user_model()
+        self.contributor = user_model.objects.create_user('test_user', 'test_pwd')
+
+    def tearDown(self):
+        for data in Data.objects.all():
+            data_dir = os.path.join(settings.FLOW_EXECUTOR['DATA_DIR'], str(data.id))
+            shutil.rmtree(data_dir, ignore_errors=True)
+
+    def test_access_api_from_docker_executor_process(self):   # pylint: disable=invalid-name
+        """Test if a process running via Docker executor can access API."""
```

Comment:
This test is practically the same as the above. How about abstracting the duplicated code? Maybe write a class method assertAccessAPI.